scene/gui/text_edit.cpp

```diff
@@ -215,9 +215,6 @@ void TextEdit::Text::_update_line_cache(int p_line) const {
 
 const Map<int, TextEdit::Text::ColorRegionInfo> &TextEdit::Text::get_color_region_info(int p_line) {
 
-	Map<int, ColorRegionInfo> *cri = NULL;
-	ERR_FAIL_INDEX_V(p_line, text.size(), *cri); //enjoy your crash
```

Comment:
I'm not sure if just removing the debugging warning here is a good idea. In release mode this crash already doesn't exist. Maybe find out what's at the other end of this and return a value that won't crash?

Reply:
Or maybe make it a fatal error rather than a warning.

Reply:
I think that by removing the explicit handling, the container implementation will crash (on purpose) on index-out-of-bounds.

Reply:
Wouldn't it be better to crash here though, for debugging purposes?

Reply:
I don't think it matters whether it crashes here or not. The backtrace will still point us to the right line.

You can try putting this in, say, the `main` function.
```cpp
String text;
text[20]; // crashes here
```

This is the `gdb` message:

```
Program received signal SIGILL, Illegal instruction.
0x00005555556e7d49 in Vector<wchar_t>::operator[] (p_index=20, this=0x7fffffffdae8) at core/vector.h:137
137                     CRASH_BAD_INDEX(p_index, size());
```

And the backtrace you will get:
```
(gdb) bt
#0  0x00005555556e7d49 in Vector<wchar_t>::operator[] (p_index=20, this=0x7fffffffdae8) at core/vector.h:137
#1  main (argc=<optimized out>, argv=<optimized out>) at platform/x11/godot_x11.cpp:48
```

It's obvious what's happening just from that, which is why I think those two lines are unnecessary.

Reply:
I think the idea is that for the end user, it's better to have the error message induced by `ERR_FAIL_INDEX_V` before a segfault that they won't know how to debug. This way they have something meaningful to report to us and not just a "it crashed".

Reply:
I'll check how Godot behaves when those lines get executed in a Clang build then and see if I can get both builds to behave the same way while keeping the messages visible.

Reply:
Changing the [for loop](https://github.com/Rubonnek/godot/blob/2b5966023f5468dd1a07b82dc2955460adc71f8a/scene/gui/text_edit.cpp#L504) to:

```cpp
for (int i = -2000; i < cursor.line_ofs + 2000; i++) {
```

to trigger `ERR_FAIL_INDEX_V` crashes both builds.

But passing a static map keeps both builds from crashing:

```cpp
	static Map<int, ColorRegionInfo> cri;
	ERR_FAIL_INDEX_V(p_line, text.size(), cri); //no more crashes on either builds
```